src/content/blog/2024-07-05-whats-coming-next-for-eslint.md

```diff
@@ -0,0 +1,57 @@
+* **Limited API.** The public API is limited because ESLint, at the start, was never intended to be used as an API. The initial `linter` object was meant only to enable a browser demo. It was later replaced by the `Linter` class, and even later, an `ESLint` class to fully mimic the command line interface. When we want to expose new functionality, it needs to exist in one of these two similar-sounding APIs, and unfortunately, the main decision point is whether or not the API needs to be available in the browser. If so, then it needs to be in `Linter`, otherwise it needs to be in `ESLint`. This is both confusing and not sustainable in the long term.
+* **Lack of type checking.** We've had the desire to add type checking into the repository to help catch potential problems, but attempts to do this in the past required a lot of work and coordination. Eventually, we came to the conclusion that it wasn't worth the effort.
+* **Stuck in CommonJS.** Similar to the type checking situation, while there is a desire to convert the codebase into ESM, the amount of work it would take to do so is prohibitive. We'd spend a lot of time on the conversion just to end up with an equivalent feature set.
+
+After thinking through the options of continually to incrementally change the existing core or starting from scratch, we [decided](https://github.com/eslint/rewrite/blob/main/decisions/001-rewrite-core.md) that it was time to start from scratch. While we firmly believe that a complete rewrite represents a significant risk, after 11 years, we can honestly say that we got the most out of the initial architecture.
+
+Going forward, here's what you can expect:
+
+* **A new repository.** The [`eslint/rewrite`](https://github.com/eslint/rewrite) repository is the new home for our language-agnostic work. It's a monorepo where we'll manage all of the packages that directly relate to the core in some way.
+* **Modern packages.** Every package in `eslint/rewrite` will be up to modern standards, publishing ESM entrypoints along with type definitions. When possible, we'll also publish CommonJS entrypoints. All packages are published with provenance to both npm and JSR (when applicable).
+* **A runtime-agnostic core.** The `@eslint/core` package will contain a runtime-agnostic API for ESLint along with type definitions for the core. We will build this package up slowly over time as we design the new API, extracting pieces of functionality out of the `eslint` package. With more JavaScript runtimes emerging each year, we feel that a runtime-agnostic core is becoming more important.
+* **A composeable API.** Instead of bundling all functionality into one or two classes, the new API will be composeable, with many purpose-built classes designed to be used together. Class will follow the single-responsibility principle, making it easier to make changes and customize ESLint inside of integrations.
```

Comment:
```suggestion
* **A composable API.** Instead of bundling all functionality into one or two classes, the new API will be composable, with many purpose-built classes designed to be used together. Class will follow the single-responsibility principle, making it easier to make changes and customize ESLint inside of integrations.
```